taichi/program/ndarray.cpp

```diff
@@ -15,18 +15,22 @@ Ndarray::Ndarray(Program *prog,
                                 std::end(shape),
                                 1,
                                 std::multiplies<>())),
-      element_size_(data_type_size(dtype)) {
+      element_size_(data_type_size(dtype)),
+      prog_impl_(prog->get_llvm_program_impl()) {
 #ifdef TI_WITH_LLVM
-  LlvmProgramImpl *prog_impl = prog->get_llvm_program_impl();
-  ndarray_alloc_ = prog_impl->allocate_memory_ndarray(nelement_ * element_size_,
-                                                      prog->result_buffer);
+  ndarray_alloc_ = prog_impl_->allocate_memory_ndarray(
+      nelement_ * element_size_, prog->result_buffer);
 
-  data_ptr_ = prog_impl->get_ndarray_alloc_info_ptr(ndarray_alloc_);
+  data_ptr_ = prog_impl_->get_ndarray_alloc_info_ptr(ndarray_alloc_);
 #else
   TI_ERROR("Llvm disabled");
 #endif
 }
 
+void Ndarray::release() {
```

Comment:
after the opengl ndarray pr is merged, this should be moved to ndarray destructor?

Reply:
yes, will do, thx